### ISSUE TITLE
[Forge] Upgrade Aquatic

### ISSUE THREAD

**Issue by riley-9054** [ZsoltMolnarrr/BetterCombat#234]
**Minecraft version** - 1.19.2
**Mod loader** - Forge 43.1.43
**Mod version** - Better Combat 1.6.2, Upgrade Aquatic 5.0.0
**I am using the latest version available** - I am using the latest 1.19.2 version of both mods at the time of writing this

**Describe the bug**
When grabbed by a thrasher from upgrade aquatic, the player loses control of their camera whenever attacking and thus cannot properly fight back.

**To Reproduce**
1. Download and install both Better Combat and Upgrade Aquatic.
2. Find and get attacked by a thrasher.
3. Attempt to swing a weapon while being attacked.
4. You will not be able to properly move your camera to face the thrasher.

**Expected behavior**
The player's camera movement not being affected at all.

**Screenshots or video recordings**
https://user-images.githubusercontent.com/117400609/230269521-4e33b376-e95e-4fb6-958e-6a22d44e6800.mp4

**Additional context**
This is almost identical to a known issue Upgrade Aquatic has with multiple other mods, mainly Optifine. To my understanding it's a result of Better Combat, Optifine, etc. messing with the code behind Minecraft's camera movement. Before going to fight thrashers I specifically uninstalled Optifine and another mod I thought would cause it called Camera Overhaul. I initially wasn't entirely positive this was caused by Better Combat but as you can see in the video i discovered that it only messes up your camera rotation while you're swinging, so it's definitely this mod.

**Comment by ZsoltMolnarrr** [comment]
This issue doesn't seem to be actionable for me.
Better Combat is not messing with camera controls.
Are you sure this is not an issue on the end of Upgrade Aquatic?

**Comment by riley-9054** [comment]
I realized a few hours after making this that it likely wouldn't be fixable on your end, but I have a feeling it has to do with Better Combat moving the player's head in attack animations without moving the first person camera. It also does not happen with just Upgrade Aquatic.
Regardless, when I encountered the issue I told the Upgrade Aquatic devs and asked them to fix it. They should be able to if Better Combat is open source, as the reason they cite for not fixing the same bug with Optifine is that Optifine is closed source. Thank you for acknowledging it at least though.

**Comment by ZsoltMolnarrr** [comment]
Better Combat does not move the camera at all.
It renders the player model while in first person (but also hides all body parts except weapons).
I also wanna note, that everything related to first person attack animations have been properly moved into PlayerAnimator library, so Better Combat no longer any code that directly influences this.

**Comment by r00sty** [comment]
So, I am trying to solve this issue for my server, I am not opposed to making a private fork of Upgrade Aquatic for my own personal use.

I looked over Upgrade Aquatic's source and it had the "shouldRiderFaceForward" value set to true, changing that to false allows the player to turn to look at it, however now when I swing a weapon affected by Better Combat, it sets my camera back to facing forward before my weapon swings.  

If I swing with an empty hand, or a block, I can hit the Thrasher with no issues.  I have added the Thrasher to the hostile_player_vehicles list, but no joy. Any ideas on what exactly could be causing this? 

I am still looking through the thrasher's code to see if there is anything that would be resetting my facing direction, but the ability to bonk it with an empty hand seems to point me back towards Better Combat.
